### PR TITLE
feat(ui): Implement Unified view search

### PR DESCRIPTION
**Pull request by bdauria**
# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/front_src/src/Resources/SearchHelpTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      <li>host.name</li>
+      <li>host.alias</li>
+      <li>host.address</li>
+      <li>service.description</li>
+    </ul>
+    <p>{`${labelSearchByHostName} host.name:hostname`}</p>
+    <p>{labelUsePartialQuery}</p>
+    <ul>
+      <li>{`${labelSearchByServiceStartingWith} service.description:centreon%`}</li>
+      <li>{`${labelSearchByHostAliasEndingWith} host.alias:%-server`}</li>
```
Comment (kduret) [inline www/front_src/src/Resources/SearchHelpTooltip.tsx]: default operator will be REGEXP, not LIKE
```suggestion
      <li>{`${labelSearchByServiceStartingWith} service.description:^centreon`}</li>
      <li>{`${labelSearchByHostAliasEndingWith} host.alias:(fr|us)-server`}</li>
```

File: www/front_src/src/Resources/api/searchObjects.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  'service.description',
+];
+
+interface SearchObject {
+  field: SearchableField;
+  value: string;
+}
+
+interface SearchParam {
+  $or: Array<{ [field in SearchableField]?: { $lk: string } }>;
```
Comment (kduret) [inline www/front_src/src/Resources/api/searchObjects.ts]:
```suggestion
  $or: Array<{ [field in SearchableField]?: { $rg: string } }>;
```

File: www/front_src/src/Resources/api/searchObjects.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const foundSearchObjects = getFoundSearchObjects(searchValue);
+
+  const searchObjectsToSend =
+    foundSearchObjects.length > 0
+      ? foundSearchObjects
+      : getDefaultSearchObjects(searchValue);
+
+  return {
+    $or: searchObjectsToSend.map(({ field, value }) => ({
+      [field]: { $lk: `${value}` },
```
Comment (kduret) [inline www/front_src/src/Resources/api/searchObjects.ts]:
```suggestion
      [field]: { $rg: `${value}` },
```

File: www/front_src/src/Resources/api/searchObjects.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  value: string;
+}
+
+interface SearchParam {
+  $or: Array<{ [field in SearchableField]?: { $lk: string } }>;
+}
+
+const getFoundSearchObjects = (searchValue: string): Array<SearchObject> => {
+  const searchOptionMatches = searchOptions.map((searchOption) => {
+    const pattern = `${searchOption.replace('.', '\\.')}:([^\\s]+)`;
```
Comment (kduret) [inline www/front_src/src/Resources/api/searchObjects.ts]: there will be search query like host.name:host_\d+
Then, need to check that this line does not break regular expression

File: www/front_src/src/Resources/translatedLabels.ts [hunk 5]
```diff
@@ -12,6 +12,21 @@ export const labelSeverity = I18n.t('Severity');
...
+  'For instance, to search by host name, you can type:',
+);
+export const labelSearchByServiceStartingWith = I18n.t(
+  "To search services with a description starting with the string 'centreon', you can type:",
+);
 export const labelState = I18n.t('State');
 export const labelStateFilter = I18n.t('State filter');
 export const labelUnhandledProblems = I18n.t('Unhandled problems');
+export const labelUsePartialQuery = I18n.t(
+  "It's also possible to use a partial query for the search value, by wrapping it into a '%' character. For instance:",
```
Comment (kduret) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
  "It's also possible to use a partial query for the search value, by using a regular expression. For instance:",
```

File: www/front_src/src/Resources/api/searchObjects.ts (hunk 4 above)
Comment (bdauria) [reply www/front_src/src/Resources/api/searchObjects.ts]: `[^\\s]+` indicates any non-whitespace character, so we're basically sending the search value as-is. This should still work with regex-like query

**Pull request merged**